validators/pool.ak

```diff
@@ -963,6 +968,73 @@ validator manage(settings_policy_id: PolicyId) {
             withdrawals,
           )
 
+        // And make sure we don't touch the assets on the pool input; they must be spent back into the same script
+        and {
+          pool_output_address == pool_input.address,
+          pool_output_value == pool_input.value,
+        }
+      }
+      AdjustLinearAmplification { pool_input } -> {
+        // Find the pool input; note that we don't look for the pool NFT here, because if someone
+        // spends with an unauthenticated UTXO, it will fail the spend script; and if someone
+        // spends with a different script address, this script can't do anything fishy,
+        // just enforces some things about the outputs
+        // This is duplicated code with the other branch, but only because we don't have pool_input yet
+        expect Some(pool_input) = list.at(inputs, pool_input)
+        let pool_input = pool_input.output
+        expect InlineDatum(datum) = pool_input.datum
+        expect datum: StablePoolDatum = datum
+        // We need the pool output to check that only the fees or fee manager are updated
+        let (
+          Output { address: pool_output_address, value: pool_output_value, .. },
+          pool_output_datum,
+        ) = find_pool_output(outputs)
+
+        let StablePoolDatum {
+          assets,
+          protocol_fees,
+          linear_amplification,
+          sum_invariant,
+          linear_amplification_manager: output_linear_amplification_manager,
+          ..
+        } = pool_output_datum
+
+        let
+          pool_quantity_a,
+          pool_quantity_b,
+        <- pool_input_to_asset_reserves(assets, protocol_fees, pool_input)
+
+        expect
+          liquidity_invariant(
+            pool_quantity_a,
+            pool_quantity_b,
+            linear_amplification,
+            sum_invariant,
+          )
+
+        let expected_datum =
+          StablePoolDatum {
+            ..datum,
+            linear_amplification: linear_amplification,
+            sum_invariant: sum_invariant,
+            linear_amplification_manager: output_linear_amplification_manager,
+          }
+        expect pool_output_datum == expected_datum
+
+        // Check that the *current* fee manager approves the update
+        expect Some(linear_amplification_manager) =
+          datum.linear_amplification_manager
+        expect
+          multisig.satisfied(
+            linear_amplification_manager,
+            extra_signatories,
+            validity_range,
+            withdrawals,
+          )
+
+        // Make sure the linear amplification is in the sane range
+        expect linear_amplification > 0 && linear_amplification <= 10000
```

Comment:
the upper bound likely needs to be much higher, and i'm not sure we should even have an upper bound :thinking:

Reply:
Let's just make sure to ask the auditors if they feel we should set an upper bound